nautobot_golden_config/models.py

```diff
@@ -177,32 +178,39 @@ def _verify_get_custom_compliance_data(compliance_details):
 
 def _get_hierconfig_remediation(obj):
     """Returns the remediating config."""
-    hierconfig_os = obj.device.platform.network_driver_mappings["hier_config"]
+    hierconfig_os = obj.device.platform.network_driver_mappings.get("hier_config")
```

Comment:
While testing the hier config v3 update, I noticed that if the `network_driver_mappings` wasn't created, because of an unsupported driver, then this would cause an exception that isn't handled. By changing this to `.get`, it produces a default value of `None`, which would cause the `ValidationError` below and handled appropriately.